gst/nnstreamer/tensor_demux/gsttensordemux.c

```diff
@@ -375,10 +394,46 @@ gst_tensor_demux_get_tensor_pad (GstTensorDemux * tensor_demux,
   g_free (stream_id);
   gst_event_unref (event);
 
-  caps = gst_tensor_caps_from_config (&config);
-  gst_pad_set_caps (pad, caps);
-  gst_element_add_pad (GST_ELEMENT_CAST (tensor_demux), pad);
+  if (tensor_demux->tensorpick != NULL) {
+    gchar *seleted_tensor;
+    gchar **strv;
+    guint num;
+    GstCaps *peer_caps;
+
+    g_assert (g_list_length (tensor_demux->tensorpick) >= nth);
+
+    seleted_tensor = (gchar *) g_list_nth_data (tensor_demux->tensorpick, nth);
+    strv = g_strsplit_set (seleted_tensor, ":+", -1);
+    num = g_strv_length (strv);
+    peer_caps = gst_pad_peer_query_caps (pad, NULL);
+
+    if (num == 1 && gst_tensor_demux_is_tensor_caps (peer_caps)) {
```

Comment:
later work, let's consider adding common function to get caps (filter / if / demux)